### PR TITLE
(spike) added user to rum

### PR DESCRIPTION
**Pull request by mdewey**
## Summary

We are looking for a way to track unique users through the system for RUM purposes. This is a spike PR to test this in staging. 

## Related issue(s)

No related issue, just slack threads.

## Testing done

- CI passes
- Need to test in staging.

# Screenshots

No UI changes

## What areas of the site does it impact?

Secure Message

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable). N/A
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] A screenshot of the developed feature is added. N/A
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed N/A

### Error Handling

- [x] The Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user. N/A

### CODE REVIEW

File: src/applications/mhv/shared/hooks/useDatadogRum.jsx [hunk 1]
```diff
@@ -20,12 +20,17 @@ const initializeDatadogRum = config => {
...
 };
 
-const useDatadogRum = config => {
+const useDatadogRum = (config, user) => {
   useEffect(
     () => {
       initializeDatadogRum(config);
+      if (user?.login?.currentlyLoggedIn) {
+        datadogRum.setUser({
+          id: user.profile.accountUUid,
```
Comment (acrollet) [inline src/applications/mhv/shared/hooks/useDatadogRum.jsx:29]: do we expect any possibility that `user.profile.accountUUid` would be missing or not set here?

File: src/applications/mhv/shared/hooks/useDatadogRum.jsx [hunk 2]
```diff
@@ -6,7 +6,7 @@ import environment from '@department-of-veterans-affairs/platform-utilities/envi
 const initializeDatadogRum = config => {
   if (
     // Prevent RUM from running on local/CI environments.
-    environment.BASE_URL.indexOf('localhost') < 0 &&
+    // environment.BASE_URL.indexOf('localhost') < 0 &&
```
Comment (acrollet) [inline src/applications/mhv/shared/hooks/useDatadogRum.jsx:9]: Just a comment that we don't want to miss reverting this line.

**Review by acrollet — DISMISSED** [review]

File: src/applications/mhv/shared/hooks/useDatadogRum.jsx (hunk 1 above)
Comment (mdewey) [reply src/applications/mhv/shared/hooks/useDatadogRum.jsx]: I don't think so, but I added in a default value to see if it ever shows up.

File: src/applications/mhv/shared/hooks/useDatadogRum.jsx [hunk 3]
```diff
@@ -20,12 +20,17 @@ const initializeDatadogRum = config => {
   }
 };
 
-const useDatadogRum = config => {
+const useDatadogRum = (config, userInfo) => {
   useEffect(
     () => {
       initializeDatadogRum(config);
+      if (userInfo.loggedIn && environment.isStaging()) {
```
Comment (mdewey) [inline src/applications/mhv/shared/hooks/useDatadogRum.jsx:27]: NOTE: if this works in, staging. We can explore adding a feature flag vs just turning it on.

**Review by acrollet — DISMISSED** [review]
re-approving 🚀

**Review by micahchiang — DISMISSED** [review]

**Review by acrollet — DISMISSED** [review]
re-approving

**Review by alex-morgun-1 — DISMISSED** [review]
great addition 👍🏻

File: src/applications/mhv/secure-messaging/containers/App.jsx [hunk 4]
```diff
@@ -81,7 +81,16 @@ const App = () => {
...
   };
-  useDatadogRum(datadogRumConfig);
+  const userDetails = useMemo(
+    () => {
+      return {
+        loggedIn: user.login.currentlyLoggedIn,
+        accountUuid: user.profile.accountUUid,
+      };
+    },
+    [user.login.currentlyLoggedIn, user.profile.accountUUid],
```
Comment (radavis) [inline src/applications/mhv/secure-messaging/containers/App.jsx:91]: Suggestion: safely navigate these object properties with the `?.` operator.

**Review by radavis — DISMISSED** [review]
Looks good. Left a suggestion.

File: src/applications/mhv/secure-messaging/containers/App.jsx (hunk 4 above)
Comment (mdewey) [reply src/applications/mhv/secure-messaging/containers/App.jsx:91]: Thats a good call.

**Review by radavis — APPROVED** [review]

**Review by acrollet — APPROVED** [review]
🐙

**Review by alex-morgun-1 — APPROVED** [review]
re-approving

**Pull request merged**